lib/internal/encoding.js

```diff
@@ -0,0 +1,375 @@
+         label[e - 1] === '\u000c' ||
+         label[e - 1] === '\u000d' ||
+         label[e - 1] === '\u0020')) {
+    e--;
+  }
+  return label.slice(s, e);
+}
+
+function getEncodingFromLabel(label) {
+  const enc = encodings.get(label);
+  if (enc !== undefined) return enc;
+  return encodings.get(trimAsciiWhitespace(label.toLowerCase()));
+}
+
+function hasTextDecoder(encoding = 'utf-8') {
+  if (typeof encoding !== 'string')
+    throw new errors.Error('ERR_INVALID_ARG_TYPE', 'encoding', 'string');
+  return hasConverter(getEncodingFromLabel(encoding));
+}
+
+class TextDecoder {
+  constructor(encoding = 'utf-8', options = {}) {
+    if (typeof encoding !== 'string')
+      throw new errors.Error('ERR_INVALID_ARG_TYPE', 'encoding', 'string');
+    if (options !== undefined && typeof options !== 'object')
+      throw new errors.Error('ERR_INVALID_ARG_TYPE', 'options', 'object');
+
+    const enc = getEncodingFromLabel(encoding);
+    if (enc === undefined || enc === 'replacement')
+      throw new errors.RangeError('ERR_ENCODING_NOT_SUPPORTED', encoding);
+
+    var flags = 0;
+    if (options !== null) {
+      flags |= options.fatal === true ? CONVERTER_FLAGS_FATAL : 0;
+      flags |= options.ignoreBOM === true ? CONVERTER_FLAGS_IGNORE_BOM : 0;
+    }
+
+    const handle = getConverter(enc, flags);
+    if (handle === undefined)
+      throw new errors.Error('ERR_ENCODING_NOT_SUPPORTED', encoding);
+
+    this[kHandle] = handle;
+    this[kFlags] = flags;
+    this[kEncoding] = enc;
+  }
+
+  get encoding() {
+    return this[kEncoding];
+  }
+
+  get fatal() {
+    return (this[kFlags] & CONVERTER_FLAGS_FATAL) === CONVERTER_FLAGS_FATAL;
+  }
+
+  get ignoreBOM() {
+    return (this[kFlags] & CONVERTER_FLAGS_IGNORE_BOM) ===
+           CONVERTER_FLAGS_IGNORE_BOM;
+  }
+
+  decode(input = empty, options = {}) {
+    if (isAnyArrayBuffer(input)) {
+      input = Buffer.from(input);
+    } else if (!ArrayBuffer.isView(input)) {
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'input',
+                                 ['ArrayBuffer', 'ArrayBufferView']);
+    }
+    if (options === null ||
+        typeof options !== 'object') {
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'options', 'object');
+    }
+
+    var flags = (options.stream === true) ? 0 : CONVERTER_FLAGS_FLUSH;
+
+    const ret = _decode(this[kHandle], input, flags);
+    if (typeof ret === 'number') {
+      const err = new errors.TypeError('ERR_ENCODING_INVALID_ENCODED_DATA',
+                                       this.encoding);
+      err.errno = ret;
+      throw err;
+    }
+    return ret.toString('ucs2');
+  }
+}
+
+var Buffer;
+function lazyBuffer() {
+  if (Buffer === undefined)
+    Buffer = require('buffer').Buffer;
+  return Buffer;
+}
+
+class TextEncoder {
+  constructor() {}
+
+  get encoding() {
+    return 'utf-8';
+  }
+
+  encode(input) {
+    return new Uint8Array(lazyBuffer().from(String(input)));
```

Comment:
You have to use template literal syntax to prevent symbols from getting converted to strings. See https://github.com/nodejs/node/blob/448c4c62d2b413226dfdef03d6f8d243de0984a3/lib/internal/url.js#L56.